kroxylicious-wasm-sample-poc/src/main/java/io/kroxylicious/sample/util/SampleWasmFilterTransformer.java

```diff
@@ -0,0 +1,131 @@
+import org.apache.kafka.common.utils.ByteBufferOutputStream;
+
+import com.dylibso.chicory.runtime.ExportFunction;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.runtime.Memory;
+import com.dylibso.chicory.runtime.Module;
+import com.dylibso.chicory.wasm.types.Value;
+
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.sample.config.SampleFilterConfig;
+
+/**
+ * Transformer class for the sample filters. Provides static transform functions for find-and-replace
+ * transformation of data in ProduceRequests and FetchResponses.
+ */
+public class SampleWasmFilterTransformer {
+
+    /**
+     * Transforms the given partition data according to the provided configuration.
+     * @param partitionData the partition data to be transformed
+     * @param context the context
+     * @param config the transform configuration
+     */
+    public static void transform(ProduceRequestData.PartitionProduceData partitionData, FilterContext context, SampleFilterConfig config) {
+        partitionData.setRecords(transformPartitionRecords((AbstractRecords) partitionData.records(), context, config.getReplacerModule()));
+    }
+
+    /**
+     * Transforms the given partition data according to the provided configuration.
+     * @param partitionData the partition data to be transformed
+     * @param context the context
+     * @param config the transform configuration
+     */
+    public static void transform(FetchResponseData.PartitionData partitionData, FilterContext context, SampleFilterConfig config) {
+        partitionData.setRecords(transformPartitionRecords((AbstractRecords) partitionData.records(), context, config.getReplacerModule()));
+    }
+
+    /**
+     * Performs find-and-replace transformations on the given partition records.
+     * @param records the partition records to be transformed
+     * @param context the context
+     * @param replacerModule the wasm replacer module to be used
+     * @return the transformed partition records
+     */
+    private static AbstractRecords transformPartitionRecords(AbstractRecords records, FilterContext context, String replacerModule) {
+        if (records.batchIterator().hasNext()) {
+            ByteBufferOutputStream stream = context.createByteBufferOutputStream(records.sizeInBytes());
+            MemoryRecordsBuilder newRecords = createMemoryRecordsBuilder(stream, records.firstBatch());
+
+            for (RecordBatch batch : records.batches()) {
+                for (Record batchRecord : batch) {
+                    newRecords.append(batchRecord.timestamp(), batchRecord.key(), transformRecord(batchRecord.value(), replacerModule),
+                            batchRecord.headers());
+                }
+            }
+            return newRecords.build();
+        }
+        return records;
+    }
+
+    // Hacked around caching of last used Wasm Module
+    private static String currentWasmModule;
```

Comment:
I get this is just a POC, but the scope looks questionable.  Perhaps scoping to the filter instance would be safer (which is effectively per connection instance).